core/ViewDataTable/Factory.php

```diff
@@ -106,12 +106,17 @@ public static function build($defaultType = null, $apiAction = false, $controlle
 
         $params = array();
 
-        if(is_null($loadViewDataTableParametersForUser)) {
+        if (!isset($loadViewDataTableParametersForUser)) {
             $loadViewDataTableParametersForUser = ('0' == Common::getRequestVar('widget', '0', 'string'));
         }
+
         if ($loadViewDataTableParametersForUser) {
             $login  = Piwik::getCurrentUserLogin();
-            $params = Manager::getViewDataTableParameters($login, $controllerAction);
+            $paramsKey = $controllerAction;
+            if (!empty($report) && $controllerAction === $apiAction) {
```

Comment:
It should be important to only apply it to the "default action" when controller and API is the same, at least for now... Not sure about implications of removing `$controllerAction === $apiAction` as it could potentially apply settings of eg a table report to a graph report...